src/ray/raylet/node_manager.cc

```diff
@@ -1050,36 +983,42 @@ void NodeManager::ProcessDisconnectClientMessage(
                    << "job_id: " << job_id;
   }
 
-  client->Close();
-
   // TODO(rkn): Tell the object manager that this client has disconnected so
   // that it can clean up the wait requests for this client. Currently I think
   // these can be leaked.
 }
 
-void NodeManager::ProcessSubmitTaskMessage(const uint8_t *message_data) {
-  // Read the task submitted by the client.
-  auto fbs_message = flatbuffers::GetRoot<protocol::SubmitTaskRequest>(message_data);
-  rpc::Task task_message;
-  RAY_CHECK(task_message.mutable_task_spec()->ParseFromArray(
-      fbs_message->task_spec()->data(), fbs_message->task_spec()->size()));
-  for (const auto &dependency :
-       string_vec_from_flatbuf(*fbs_message->execution_dependencies())) {
-    task_message.mutable_task_execution_spec()->add_dependencies(dependency);
+/// Handle a `SubmitTask` request.
+void NodeManager::HandleSubmitTaskRequest(const rpc::SubmitTaskRequest &request,
+                                          rpc::SubmitTaskReply *reply,
+                                          rpc::SendReplyCallback send_reply_callback) {
+  RAY_LOG(DEBUG) << "Handle submit task request.";
```

Comment:
at `task_id` into the log? otherwise we're not able to distinguish which task it is.

Reply:
It seems there is a very detailed log is in SubmitTask function